pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,481 @@
+
+import numpy as np
+from pgmpy.models import BayesianModel
+from pgmpy.factors import TabularCPD, TreeCPD, RuleCPD
+from collections import OrderedDict, namedtuple
+
+
+edge_type = namedtuple('edge_type', ('start', 'end'))
+
+
+class DynamicBayesianNetwork(BayesianModel):
+    def __init__(self, ebunch=None):
+        """
+        Base class for Dynamic Bayesian Network
+
+        This model is a time variant of the static Bayesian model, where each timeslice has
+        some static nodes and is then replicated over a certain time slice.
+
+        The nodes can be hashable python objects.
+
+        However, the hidden nodes will compulsory have the following form.
+        (node_name, time_slice)
+
+        Here, node_name is the node that is inserted
+        while the time_slice is an integer value, which denotes
+        the index of the time_slice that the node belongs to.
+
+        Edges are represented as links between the nodes.
+
+        Parameters:
+        ----------
+        data: Data to initialize graph.  If data=None (default) an empty
+              graph is created.  The data can be an edge list, or any
+              NetworkX graph object
+
+        Examples:
+        --------
+        Create an empty Dynamic Bayesian Network with no nodes and no edges
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        adding nodes and edges inside the dynamic bayesian network. A single
```

Comment:
@palashahuja Make the docs more clear.. Add a space above this line and use capital a in adding.